hbase_sql/src/main/java/com/splicemachine/derby/lifecycle/HEngineSqlEnv.java

```diff
@@ -265,21 +270,38 @@ public int getMaxExecutorCores() {
         if (MAX_EXECUTOR_CORES != -1)
             return MAX_EXECUTOR_CORES;
         synchronized (HEngineSqlEnv.class) {
-            int sparkNodes = getNumSparkNodes();
+            byte [] sparkYarnConfigBytes = getSparkYarnConfigBytes();
+            if (sparkYarnConfigBytes == null)
+                return DEFAULT_MAX_EXECUTOR_CORES;
+
+            SparkYarnConfiguration conf = null;
+            try {
+                ByteArrayInputStream bis = new ByteArrayInputStream(sparkYarnConfigBytes);
+                ObjectInput in = new ObjectInputStream(bis);
+                conf = (SparkYarnConfiguration) in.readObject();
+            }
+            catch (Exception e) {
+                return DEFAULT_MAX_EXECUTOR_CORES;
+            }
+            if (conf == null)
+                return DEFAULT_MAX_EXECUTOR_CORES;
+
             int maxExecutorCores =
-              calculateMaxExecutorCores(HConfiguration.unwrapDelegate().get("yarn.nodemanager.resource.memory-mb"),
-                                        getProperty("splice.spark.dynamicAllocation.enabled"),
-                                        getProperty("splice.spark.executor.instances"),
-                                        getProperty("splice.spark.executor.cores"),
-                                        getProperty("splice.spark.executor.memory"),
-                                        getProperty("splice.spark.dynamicAllocation.maxExecutors"),
-                                        getProperty("splice.spark.executor.memoryOverhead"),
-                                        getProperty("splice.spark.yarn.executor.memoryOverhead"),
-                                        sparkNodes > 0 ? sparkNodes : 1);
-            if (sparkNodes > 0) {
-                numSparkNodes = sparkNodes;
+              calculateMaxExecutorCores(conf.getYarnNodemanagerResourceMemoryMB(),
```

Comment:
I'd rather have this function in OlapServerMaster or a class logically close to the OlapServer, I think this class shouldn't have that much knowledge about YARN. Ideally we should just read the number of executor cores from Zookeeper.

Reply:
OK, I moved the calculation to OlapServerMaster.